src/media_session.rs

```diff
@@ -0,0 +1,106 @@
+use crate::cleanup_timespan::Cleanup;
+use crate::MediaStatus;
+use human_repr::HumanDurationData;
+use std::fmt;
+use windows::Media::Control::{
+    GlobalSystemMediaTransportControlsSession, GlobalSystemMediaTransportControlsSessionManager,
+    GlobalSystemMediaTransportControlsSessionMediaProperties,
+    GlobalSystemMediaTransportControlsSessionTimelineProperties,
+};
+
+pub struct MediaSession {
+    session: GlobalSystemMediaTransportControlsSession,
+    properties: GlobalSystemMediaTransportControlsSessionMediaProperties,
+    timeline: GlobalSystemMediaTransportControlsSessionTimelineProperties,
+}
+
+impl MediaSession {
+    pub async fn new() -> Result<Self, windows::core::Error> {
+        let mp = GlobalSystemMediaTransportControlsSessionManager::RequestAsync()?.await?;
+        let session = mp.GetCurrentSession()?;
+        let properties = session.TryGetMediaPropertiesAsync()?.await?;
+        let timeline = session.GetTimelineProperties()?;

```

Comment:
These lines should be moved into respective functions. They are unnecessary WinRT api calls.

Reply:
how about I turn the properties of `MediaSession` for those two to be optional, so that if, `get_artist` is called, then `get_title` is called, the api will only be called once? (this is how it's currently done for `fmt` attribute of `MediaSession`